cpp/src/arrow/util/byte_stream_split.h

```diff
@@ -25,23 +25,44 @@
 
 #ifdef ARROW_HAVE_SSE4_2
 // Enable the SIMD for ByteStreamSplit Encoder/Decoder
-#define ARROW_HAVE_SIMD_SPLIT
+#define ARROW_HAVE_SIMD_DECODE_SPLIT
+#define ARROW_HAVE_SIMD_ENCODE_SPLIT
 #endif  // ARROW_HAVE_SSE4_2
 
+#ifdef ARROW_HAVE_NEON
+#define ARROW_HAVE_SIMD_DECODE_SPLIT
+#endif  // ARROW_HAVE_NEON
+
 namespace arrow {
 namespace util {
 namespace internal {
 
-#if defined(ARROW_HAVE_SSE4_2)
+#if defined(ARROW_HAVE_NEON)
+#define DATA_TYPE_128BIT uint8x16_t
+#define LDR_SIMD_128BIT vld1q_u8
+#define STR_SIMD_128BIT vst1q_u8
+#define REINT_CAST_TYPE uint8_t
+#define SHUFFLE_LO_128BIT vzip1q_u8
+#define SHUFFLE_HI_128BIT vzip2q_u8
+#elif defined(ARROW_HAVE_SSE4_2)
```

Comment:
An SIMD wrapper lib looks better than doing this ourselves. And I think there are other SSE4 codes in Arrow have similar NEON implementation.
@pitrou @jianxind

Reply:
I agree that we should discuss adopting an SIMD library. There are several of them.
https://github.com/agenium-scale/nsimd
https://github.com/aff3ct/MIPP
https://xsimd.readthedocs.io/en/latest/
https://github.com/p12tic/libsimdpp

Do you want to start the discussion on the mailing-list @cyb70289 ?

Reply:
I've sent a mail to dev mailing-list for discussion.